pinot-tools/src/main/java/com/linkedin/pinot/tools/ConfigValidator.java

```diff
@@ -0,0 +1,129 @@
+ * <p>No guarantee on config's content.
+ */
+public class ConfigValidator {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigValidator.class);
+
+  private static final HttpClient HTTP_CLIENT = new HttpClient();
+
+  private final String _controllerHost;
+  private final int _controllerPort;
+
+  public ConfigValidator(String controllerHost, int controllerPort) {
+    _controllerHost = controllerHost;
+    _controllerPort = controllerPort;
+  }
+
+  public void validateTableConfig() {
+    try {
+      URL url = new URL("http", _controllerHost, _controllerPort, "/tables");
+      GetMethod httpGet = new GetMethod(url.toString());
+      int responseCode = HTTP_CLIENT.executeMethod(httpGet);
+      String response = httpGet.getResponseBodyAsString();
+      if (responseCode != 200) {
+        LOGGER.error("While fetching table names, got error response code: {}, response: {}", responseCode, response);
+      } else {
+        JSONArray tables = new JSONObject(response).getJSONArray("tables");
+        int numTables = tables.length();
+        for (int i = 0; i < numTables; i++) {
+          String table = tables.getString(i);
+          LOGGER.info("Validating table config for table: \"{}\"", table);
+          try {
+            url = new URL("http", _controllerHost, _controllerPort, "/tables/" + table);
+            httpGet = new GetMethod(url.toString());
+            HTTP_CLIENT.executeMethod(httpGet);
+            responseCode = HTTP_CLIENT.executeMethod(httpGet);
+            if (responseCode != 200) {
+              response = httpGet.getResponseBodyAsString();
+              throw new RuntimeException("Got error response code: " + responseCode + ", response: " + response);
+            }
+          } catch (Exception e) {
+            LOGGER.error("Table config validation failed for table: \"{}\"", table, e);
+          }
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while validating table config for controller: {}, port: {}", _controllerHost,
+          _controllerPort, e);
+    }
+  }
+
+  public void validateSchema() {
+    try {
+      URL url = new URL("http", _controllerHost, _controllerPort, "/schemas");
+      GetMethod httpGet = new GetMethod(url.toString());
+      int responseCode = HTTP_CLIENT.executeMethod(httpGet);
+      String response = httpGet.getResponseBodyAsString();
+      if (responseCode != 200) {
+        LOGGER.error("While fetching schema names, got error response code: {}, response: {}", responseCode, response);
+      } else {
+        JSONArray schemas = new JSONArray(response);
+        int numSchemas = schemas.length();
+        for (int i = 0; i < numSchemas; i++) {
+          String schema = schemas.getString(i);
+          LOGGER.info("Validating schema: \"{}\"", schema);
+          try {
+            url = new URL("http", _controllerHost, _controllerPort, "/schemas/" + schema);
```

Comment:
Use a central place to build this path (eg ControllerRequestBuilder, or something similar), instead of assuming the url here.